ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -398,7 +398,24 @@ private boolean deleteIfNeeded(BlobId blobId, long deletionTime) throws CloudSto
 
   @Override
   public short undelete(MessageInfo info) throws StoreException {
-    throw new UnsupportedOperationException("Undelete not supported in cloud store");
+    // TODO: Currently this is implemented for undeletes via replication only for DR.
```

Comment:
BTW, delete method would also have to be take lifeVersion into consideration. For instance, if the current lifeVersion is 0 and replication thread calls delete with lifeVersion being 2, then we have to update lifeVersion as well, even if we don't change the deletion_time.